x-pack/platform/plugins/shared/ml/server/saved_objects/saved_objects.ts

```diff
@@ -27,13 +29,38 @@ interface MlModuleAttributes {
   datafeeds: object[];
 }
 
+const AD_PREFIX = `${AD_SAVED_OBJECT_TYPE}-`;
+const DFA_PREFIX = `${DFA_SAVED_OBJECT_TYPE}-`;
+
 export function setupSavedObjects(savedObjects: SavedObjectsServiceSetup) {
   savedObjects.registerType({
     name: ML_JOB_SAVED_OBJECT_TYPE,
     hidden: false,
     namespaceType: 'multiple',
     migrations,
     mappings: mlJob,
+    management: {
+      importableAndExportable: true,
+      getTitle(obj) {
+        return obj.id;
+      },
+      getInAppUrl(obj) {
+        if (obj.id.startsWith(AD_PREFIX)) {
+          const jobId = obj.id.slice(AD_PREFIX.length);
+          return {
+            path: `/app/management/insightsAndAlerting/jobsListLink/${jobId}`,
+            uiCapabilitiesPath: 'ml.canGetJobs',
```

Comment:
Is it possible to add additional capabilities checks to control what actions a user can perform and whether or not the saved objects are listed?

I think should not list the SOs if the user does not have the minimum read permissions and should only be able to edit them if they have write permissions, e.g. for AD SOs:
`canGetJobs` AD job saved objects are listed but are read only.
`canGetJobs && canCreateJobs`  AD job saved objects are listed and can be edited

Reply:
Thanks! Added capabilities check [here](https://github.com/elastic/kibana/pull/205177/commits/1450c34fb4dd317728d0807711664f68e28da4f4) and [here](https://github.com/elastic/kibana/pull/205177/commits/ffc1890c98257ae73491d978f70ab33c2b0288a7)